cc.c

```diff
@@ -47,6 +47,9 @@ int main(int argc, char** argv)
 	FILE* destination_file = stdout;
 	Architecture = 0; /* catch unset */
 	init_macro_env("__M2__", "42", "__INTERNAL_M2__", 0); /* Setup __M2__ */
+	init_macro_env("__M2C__", "1", "__INTERNAL_M2__", 0);
```

Comment:
Hmm, not against this but wouldn't something like
`__M2C_MAJOR__`, `__M2C_MINOR__`, `__M2C_PATCHLEVEL__`
or maybe
`__M2_MAJOR__`, `__M2_MINOR__`, `__M2_PATCHLEVEL__`
be better.

Reply:
I'm not sure. It's based on GCC https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html which has the `__GNUC__` `__GNUC_MINOR__`, and `__GNUC_PATCHLEVEL__` defines. I don't really have an opinion, but I think there's value in mirroring the standard even if it doesn't really make sense. Let me know what you think and I'll change it if it's important.